File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023/days/Day13Test.java
package com.capital7software.aoc.aoc2023.days;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.logging.Logger;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day13Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = Logger.getLogger(Day13Test.class.getName());

    @BeforeEach
    void setUp() {
        var instance = new Day13();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void summarizePatternReflectionNotes() {
        var instance = new Day13();
        int expected = 405;
        var actual = instance.summarizePatternReflectionNotes(lines, 0);
        assertEquals(expected, actual, "The pattern notes summary " +
                "is not what was expected: " + expected);
    }

    @Test
    void summarizePatternReflectionNotesWithSmudge() {
        var instance = new Day13();
        int expected = 400;
        var actual = instance.summarizePatternReflectionNotes(lines, 1);
        assertEquals(expected, actual, "The pattern notes counting smudges summary " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2024/src/test/kotlin/com/capital7software/aoc/aoc2024aoc/days/Day13Test.kt
package com.capital7software.aoc.aoc2024aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day13Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day13Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day13()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testFewestTokensToWinAllPrizes() {
    val instance = Day13()
    val expected = 480L

    val actual = instance.fewestTokensToWinAllPrizes(lines)
    Assertions.assertEquals(
        expected,
        actual,
        "The fewest tokens of $actual is not the expected fewest tokens of $expected."
    )
  }

  @Test
  fun testFewestTokensToWinAllPrizesWithOffset() {
    val instance = Day13()
    val expected = 875_318_608_908

    val actual = instance.fewestTokensToWinAllPrizesWithOffset(lines)
    Assertions.assertEquals(
        expected,
        actual,
        "The fewest tokens of $actual is not the expected fewest tokens of $expected."
    )
  }
}
